src/components/text-editor/text-editor.tsx

```diff
@@ -0,0 +1,98 @@
+import { Component, Element, Prop, forceUpdate, h } from '@stencil/core';
+import { Editor, EditorProps } from '@toast-ui/react-editor';
+import { EditorType } from '@toast-ui/editor/types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import '@toast-ui/editor/dist/toastui-editor.css';
+
+/**
+ * The `limel-text-editor` component wraps the toast-ui react editor. This editor
+ * allows a rich text editing experience with markdown support.
+ */
+@Component({
+    tag: 'limel-text-editor',
+    shadow: true,
+    styleUrl: 'text-editor.scss',
+})
+export class TextEditor {
```

Comment:
To make sure this is compatible to be used in a form, it should implement the `FormComponent<string>` interface